# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1450307499 18000
#      Wed Dec 16 18:11:39 2015 -0500
# Node ID cfb58d9805fa72979780fa1185e4a005458eea05
# Parent  6f0bd96f93c0b3423feebe7c8fe7843a362f9ede
eliminate some type conversion warnings

* oct-rl-edit.h, oct-rl-edit.c (octave_rl_add_defun): Change prototype
to match rl_add_defun.
(octave_rl_ctrl, octave_rl_meta): Return int instead of char.
* oct-rl-hist.c (octave_history_list): Use size_t as needed.

diff --git a/liboctave/util/oct-rl-edit.c b/liboctave/util/oct-rl-edit.c
--- a/liboctave/util/oct-rl-edit.c
+++ b/liboctave/util/oct-rl-edit.c
@@ -402,17 +402,17 @@ octave_rl_prompt_start_ignore (void)
 
 char
 octave_rl_prompt_end_ignore (void)
 {
   return RL_PROMPT_END_IGNORE;
 }
 
 void
-octave_rl_add_defun (const char *name, rl_fcn_ptr f, char key)
+octave_rl_add_defun (const char *name, rl_fcn_ptr f, int key)
 {
   rl_add_defun (name, f, key);
 }
 
 void
 octave_rl_set_terminal_name (const char *term)
 {
   OCTAVE_RL_SAVE_STRING (saved_term, term);
@@ -437,21 +437,21 @@ octave_rl_history_search_forward (int co
 }
 
 int
 octave_rl_history_search_backward (int count, int ignore)
 {
   return rl_history_search_backward (count, ignore);
 }
 
-char
+int
 octave_rl_ctrl (char c)
 {
   return CTRL (c);
 }
 
-char
+int
 octave_rl_meta (char c)
 {
   return META (c);
 }
 
 #endif
diff --git a/liboctave/util/oct-rl-edit.h b/liboctave/util/oct-rl-edit.h
--- a/liboctave/util/oct-rl-edit.h
+++ b/liboctave/util/oct-rl-edit.h
@@ -149,27 +149,27 @@ extern rl_event_hook_fcn_ptr octave_rl_g
 
 extern char **
 octave_rl_completion_matches (const char *, rl_completer_fcn_ptr);
 
 extern char octave_rl_prompt_start_ignore (void);
 
 extern char octave_rl_prompt_end_ignore (void);
 
-extern void octave_rl_add_defun (const char *, rl_fcn_ptr, char);
+extern void octave_rl_add_defun (const char *, rl_fcn_ptr, int);
 
 extern void octave_rl_set_terminal_name (const char *);
 
 extern void octave_rl_initialize (void);
 
 extern int octave_rl_history_search_forward (int, int);
 
 extern int octave_rl_history_search_backward (int, int);
 
-extern char octave_rl_ctrl (char);
+extern int octave_rl_ctrl (char);
 
-extern char octave_rl_meta (char);
+extern int octave_rl_meta (char);
 
 #ifdef __cplusplus
 }
 #endif
 
 #endif
diff --git a/liboctave/util/oct-rl-hist.c b/liboctave/util/oct-rl-hist.c
--- a/liboctave/util/oct-rl-hist.c
+++ b/liboctave/util/oct-rl-hist.c
@@ -274,23 +274,23 @@ octave_history_list (int limit, int numb
 
       int beg = 0;
       int end = 0;
       while (hlist[end])
         end++;
 
       beg = (limit < 0 || end < limit) ? 0 : (end - limit);
 
-      retval = malloc ((end - beg + 1) * sizeof (char **));
+      retval = malloc ((size_t) (end - beg + 1) * sizeof (char **));
 
       k = 0;
       for (i = beg; i < end; i++)
         {
           char *line = hlist[i]->line;
-          int len = line ? strlen (line) : 0;
+          size_t len = line ? strlen (line) : 0;
           char *tmp = malloc (len + 64);
 
           if (number_lines)
             sprintf (tmp, "%5d %s", i + history_base,
                      line ? line : "");
           else
             strcpy (tmp, line ? line : "");
 
